File: card.go
// The model for a single playing card in Whist
//
// Ranking is
// A (14)
// K (13)
// Q (12)
// J (11)
// 10
// ..
// 1

package main

import (
	"fmt"
	"strconv"
)

type card struct {
	rank int
	suit string
}

func (card card) friendlyName() string {
	var cardName string

	switch card.rank {
	case 11:
		cardName = "Jack"
	case 12:
		cardName = "Queen"
	case 13:
		cardName = "King"
	case 14:
		cardName = "Ace"
	default:
		cardName = strconv.Itoa(card.rank)
	}

	return fmt.Sprintf("The %s of %s", cardName, card.suit)
}

File: lib/card.rb
class Card
  def initialize(rank, suit)
    @rank = rank
    @suit = suit
  end

  def rank
    @rank
  end

  def suit
    @suit
  end

  def friendly_name
    friendly_name = ""
    case @rank
    when 11
      friendly_name = "Jack"
    when 12
      friendly_name = "Queen"
    when 13
      friendly_name = "King"
    when 14
      friendly_name = "ace"
    else
      friendly_name = @rank.to_s
    end

    "The #{friendly_name} of #{self.suit}"
  end
end